### PR TITLE
Ruby on Rails Capstone Project :gem: 

### PR DESCRIPTION
**Pull request by sergio-usma**
# Ruby on Rails Capstone Project :gem: 

This project addresses the requirements outlined in the Ruby on Rails capstone project. The changes include the implementation of user authentication, transaction management, and category creation functionalities in the mobile web application. Additionally, the design guidelines and technical specifications have been adhered to, ensuring a professional and secure application.

### User Authentication and Registration
- [X] Implemented user registration and login functionality.
- [X] Enforced mandatory fields for full name, email, and password during the registration process.

### Home Page (Categories Page)
- [X] Designed a splash screen with the app name, providing links to the sign-up and login pages.
- [X] Implemented the categories page, displaying category names, icons, and total transaction amounts.
- [X] Added functionality to navigate to the transactions page for a selected category.
- [X] Included a "Add a new category" button for creating new categories.
### Transactions Page
- [X] Displayed a list of transactions for a selected category, ordered by the most recent.
- [X] Showcased the total amount for the category at the top of the page.
- [X] Implemented a "Add a new transaction" button to navigate to the transaction creation page.
- [X] Enabled navigation to the home page via the "Back" button.
### "Add a new category" Page
- [X] Created a form for users to input category details, including name and icon.
- [X] Implemented functionality to save the new category and navigate to the home page upon success.
- [X] Provided navigation to the home page via the "Back" button.
### "Add a new transaction" Page
- [X] Developed a form for users to input transaction details, including name, amount, and categories.
- [X] Implemented functionality to save the new transaction and navigate to the transactions page for that category.
- [X] Enabled navigation to the transactions page for that category via the "Back" button.
### Testing
- [X] Implemented unit and integration tests for critical components of the Ruby on Rails application.
## Technical Specifications
- [X] Utilized Postgres as the database.
- [X] Incorporated Devise for user authentication.
- [X] Validated user input to enhance app security.
- [X] Employed the .erb view template engine.
- [X] Deployed the project, making it accessible online.
## Additional Information
- [X] Followed design guidelines, including color, typography, and layout specifications.
- [X] Provided appropriate credit to the original design author, Gregoire Vella, in the README file.
## Testing Results
All implemented functionalities have been thoroughly tested, and unit/integration tests have been added to ensure the robustness and reliability of the application.
image
## Screenshots

| Login                                                            | Features                                                         |
|------------------------------------------------------------------|------------------------------------------------------------------|
| mobile | mobile (2) |
| mobile (1) | mobile (3) |

### Presentation video

<div>
    <a href="https://www.loom.com/share/7cd85feb2f884628b300696cd9badd49">
      <p>Rails capstone project presentation - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/7cd85feb2f884628b300696cd9badd49">
      
    </a>
  </div>

## Checklist
- [X] No linter errors present.
- [X] Correct gitflow has been followed.
- [X] Professional documentation has been maintained.
## Note to Reviewers
Kindly review the changes and provide feedback. The pull request adheres to the specified project requirements and enhances the overall functionality and security of the Ruby on Rails application.

🚀 Thank you for your time and consideration! 🚀

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +56,14 @@ This BudgetApp is a Ruby on Rails 7 project that helps you manage your expenses
 - [Click here](https://budget-app-5bnf.onrender.com/) to see the like demo!
```
Comment (aamirkhan2478) [inline README.md]: - [x] Well done :clap: your project is very nice :+1: but your live version is not working when I click on the link you provided for the live demo it shows a `not found` error so kindly add the working link :slightly_smiling_face:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @sergio-usma, 



Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Good Effort
✔️ Your design is good.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (sergio-usma) [reply README.md]: You're right! I have updated the link. Thanks!

**Review by fobadara — APPROVED** [review]
Hi @sergio-usma, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good use of separation of concerns


## Optional suggestions
I am including these suggestions as optional even though they should be required changes because the previous reviewer missed them.
[OPTIONAL] I get an error after signup even though the signup works
https://github.com/sergio-usma/budget-app.git

[OPTIONAL] I am asked to confirm that I signedup even though I don't receive any email regarding this.


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**